acnportal/acnsim/simulator.py

```diff
@@ -210,6 +227,170 @@ def index_of_evse(self, station_id):
+        after the Simulator is loaded.
+
+        """
+        context_dict, loaded_dict, cls_kwargs = \
+            base.none_to_empty_dict(context_dict, loaded_dict, cls_kwargs)
+
+        network = base.read_from_id(in_dict['network'], context_dict,
+                                    loaded_dict=loaded_dict)
+
+        events = base.read_from_id(in_dict['event_queue'],
+                                   context_dict=context_dict,
+                                   loaded_dict=loaded_dict)
+
+        scheduler_cls = base.locate(in_dict['scheduler'])
+        try:
+            scheduler = scheduler_cls()
+        except TypeError:
+            warnings.warn(f"Scheduler {in_dict['scheduler']} requires "
+                          f"constructor inputs. Setting scheduler to "
+                          f"BaseAlgorithm instead.")
+            scheduler = BaseAlgorithm()
+
+        if sys.version_info[1] < 7:
+            warnings.warn(f"ISO format {in_dict['start']} cannot be loaded as "
+                          f"datetime object. Use python 3.7 or higher to load "
+                          f"this value.")
+            start = in_dict['start']
+        else:
+            start = datetime.fromisoformat(in_dict['start'])
+
+        out_obj = cls(
+            network,
+            scheduler,
+            events,
+            start,
+            period=in_dict['period'],
+            signals=in_dict['signals'],
+            verbose=in_dict['verbose'],
+            **cls_kwargs
+        )
+        scheduler.register_interface(Interface(out_obj))
+
+        attr_lst = ['max_recompute', 'peak', '_iteration', '_resolve',
+                    '_last_schedule_update']
+        for attr in attr_lst:
+            setattr(out_obj, attr, in_dict[attr])
+
+        if in_dict['schedule_history'] is not None:
+            out_obj.schedule_history = {int(key): value
+                                        for key, value
+                                        in in_dict['schedule_history'].items()}
+        else:
+            out_obj.schedule_history = None
+
+        out_obj.pilot_signals = np.array(in_dict['pilot_signals'])
+        out_obj.charging_rates = np.array(in_dict['charging_rates'])
+
+        out_obj.ev_history = {session_id: base.read_from_id(
+            ev, context_dict=context_dict, loaded_dict=loaded_dict)
+            for session_id, ev in in_dict['ev_history'].items()}
+
+        out_obj.event_history = [base.read_from_id(
+            past_event, context_dict=context_dict, loaded_dict=loaded_dict)
+            for past_event in in_dict['event_history']]
+        return out_obj
+
+    def update_scheduler(self, new_scheduler):
+        """ Updates a Simulator's schedule. """
+        self.scheduler = new_scheduler
+        self.scheduler.register_interface(Interface(self))
+        self.max_recompute = new_scheduler.max_recompute
+
+    def update_signals(self, new_signals):
```

Comment:
Consider removing this since we can set it directly.

Reply:
Done.